File: Week6/gregsListVue/src/models/House.js


export class House {
	constructor (data) {
		this.id = data.id
		this.bedrooms = data.bedrooms
		this.bathrooms = parseFloat(data.bathrooms)
		this.imgUrl = data.imgUrl
		this.levels = data.levels
		this.price = data.price
		this.year = data.year
		this.description = data.description
		this.creatorId = data.creatorId
		this.createdAt = new Date(data.createdAt)
		this.updatedAt = new Date(data.updatedAt)
		this.creator = data.creator
	}
}

File: Week10/gregsListCSharp/server/Models/House.cs
namespace csharp_gregslist_api.Models;

public class House
{

	// sqft,
	// bedrooms,
	// bathrooms,
	// imgUrl,
	// description,
	// price,
	// creatorId
	public int Id { get; set; }
	public DateTime CreatedAt { get; set; }
	public DateTime UpdatedAt { get; set; }
	public int? Sqft { get; set; }
	public int? Bedrooms { get; set; }
	public double? Bathrooms { get; set; }
	// if price is not present in request body, default to null instead of 0
	public string ImgUrl { get; set; }
	public string Description { get; set; }
	public int? Price { get; set; }
	public string CreatorId { get; set; }

	public Account Creator { get; set; }
}